// @kinde-oss/kinde-typescript-sdk/dist/models/CreatePropertyResponseProperty.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the CreatePropertyResponseProperty interface.
 */
export function instanceOfCreatePropertyResponseProperty(value) {
    var isInstance = true;
    return isInstance;
}
export function CreatePropertyResponsePropertyFromJSON(json) {
    return CreatePropertyResponsePropertyFromJSONTyped(json, false);
}
export function CreatePropertyResponsePropertyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}
export function CreatePropertyResponsePropertyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
    };
}
