// @kinde-oss/kinde-typescript-sdk/dist/models/CreatePropertyResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { CreatePropertyResponsePropertyFromJSON, CreatePropertyResponsePropertyToJSON, } from './CreatePropertyResponseProperty.js';
/**
 * Check if a given object implements the CreatePropertyResponse interface.
 */
export function instanceOfCreatePropertyResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function CreatePropertyResponseFromJSON(json) {
    return CreatePropertyResponseFromJSONTyped(json, false);
}
export function CreatePropertyResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !exists(json, 'message') ? undefined : json['message'],
        'code': !exists(json, 'code') ? undefined : json['code'],
        'property': !exists(json, 'property') ? undefined : CreatePropertyResponsePropertyFromJSON(json['property']),
    };
}
export function CreatePropertyResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'code': value.code,
        'property': CreatePropertyResponsePropertyToJSON(value.property),
    };
}
