// @kinde-oss/kinde-typescript-sdk/dist/models/CreateCategoryRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * @export
 */
export var CreateCategoryRequestContextEnum = {
    Org: 'org',
    Usr: 'usr'
};
/**
 * Check if a given object implements the CreateCategoryRequest interface.
 */
export function instanceOfCreateCategoryRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "context" in value;
    return isInstance;
}
export function CreateCategoryRequestFromJSON(json) {
    return CreateCategoryRequestFromJSONTyped(json, false);
}
export function CreateCategoryRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'context': json['context'],
    };
}
export function CreateCategoryRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'context': value.context,
    };
}
