// @kinde-oss/kinde-typescript-sdk/dist/models/CreateCategoryResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { CreateCategoryResponseCategoryFromJSON, CreateCategoryResponseCategoryToJSON, } from './CreateCategoryResponseCategory.js';
/**
 * Check if a given object implements the CreateCategoryResponse interface.
 */
export function instanceOfCreateCategoryResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function CreateCategoryResponseFromJSON(json) {
    return CreateCategoryResponseFromJSONTyped(json, false);
}
export function CreateCategoryResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !exists(json, 'message') ? undefined : json['message'],
        'code': !exists(json, 'code') ? undefined : json['code'],
        'category': !exists(json, 'category') ? undefined : CreateCategoryResponseCategoryFromJSON(json['category']),
    };
}
export function CreateCategoryResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'code': value.code,
        'category': CreateCategoryResponseCategoryToJSON(value.category),
    };
}
