// @kinde-oss/kinde-typescript-sdk/dist/models/GetCategoriesResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { CategoryFromJSON, CategoryToJSON, } from './Category.js';
/**
 * Check if a given object implements the GetCategoriesResponse interface.
 */
export function instanceOfGetCategoriesResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetCategoriesResponseFromJSON(json) {
    return GetCategoriesResponseFromJSONTyped(json, false);
}
export function GetCategoriesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'categories': !exists(json, 'categories') ? undefined : (json['categories'].map(CategoryFromJSON)),
        'hasMore': !exists(json, 'has_more') ? undefined : json['has_more'],
    };
}
export function GetCategoriesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'categories': value.categories === undefined ? undefined : (value.categories.map(CategoryToJSON)),
        'has_more': value.hasMore,
    };
}
