// @kinde-oss/kinde-typescript-sdk/dist/models/GetPropertyValuesResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { PropertyValueFromJSON, PropertyValueToJSON, } from './PropertyValue.js';
/**
 * Check if a given object implements the GetPropertyValuesResponse interface.
 */
export function instanceOfGetPropertyValuesResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetPropertyValuesResponseFromJSON(json) {
    return GetPropertyValuesResponseFromJSONTyped(json, false);
}
export function GetPropertyValuesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'properties': !exists(json, 'properties') ? undefined : (json['properties'].map(PropertyValueFromJSON)),
        'nextToken': !exists(json, 'next_token') ? undefined : json['next_token'],
    };
}
export function GetPropertyValuesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'properties': value.properties === undefined ? undefined : (value.properties.map(PropertyValueToJSON)),
        'next_token': value.nextToken,
    };
}
