// @kinde-oss/kinde-typescript-sdk/dist/models/GetPropertiesResponse.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
import { PropertyFromJSON, PropertyToJSON, } from './Property.js';
/**
 * Check if a given object implements the GetPropertiesResponse interface.
 */
export function instanceOfGetPropertiesResponse(value) {
    var isInstance = true;
    return isInstance;
}
export function GetPropertiesResponseFromJSON(json) {
    return GetPropertiesResponseFromJSONTyped(json, false);
}
export function GetPropertiesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !exists(json, 'code') ? undefined : json['code'],
        'message': !exists(json, 'message') ? undefined : json['message'],
        'properties': !exists(json, 'properties') ? undefined : (json['properties'].map(PropertyFromJSON)),
        'hasMore': !exists(json, 'has_more') ? undefined : json['has_more'],
    };
}
export function GetPropertiesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'properties': value.properties === undefined ? undefined : (value.properties.map(PropertyToJSON)),
        'has_more': value.hasMore,
    };
}
