// @kinde-oss/kinde-typescript-sdk/dist/models/Property.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the Property interface.
 */
export function instanceOfProperty(value) {
    var isInstance = true;
    return isInstance;
}
export function PropertyFromJSON(json) {
    return PropertyFromJSONTyped(json, false);
}
export function PropertyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !exists(json, 'id') ? undefined : json['id'],
        'key': !exists(json, 'key') ? undefined : json['key'],
        'name': !exists(json, 'name') ? undefined : json['name'],
        'isPrivate': !exists(json, 'is_private') ? undefined : json['is_private'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'isKindeProperty': !exists(json, 'is_kinde_property') ? undefined : json['is_kinde_property'],
    };
}
export function PropertyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
        'name': value.name,
        'is_private': value.isPrivate,
        'description': value.description,
        'is_kinde_property': value.isKindeProperty,
    };
}
