// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateOrganizationPropertiesRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
/**
 * Check if a given object implements the UpdateOrganizationPropertiesRequest interface.
 */
export function instanceOfUpdateOrganizationPropertiesRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "properties" in value;
    return isInstance;
}
export function UpdateOrganizationPropertiesRequestFromJSON(json) {
    return UpdateOrganizationPropertiesRequestFromJSONTyped(json, false);
}
export function UpdateOrganizationPropertiesRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'properties': json['properties'],
    };
}
export function UpdateOrganizationPropertiesRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'properties': value.properties,
    };
}
