// @kinde-oss/kinde-typescript-sdk/dist/models/UpdatePropertyRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdatePropertyRequest interface.
 */
export function instanceOfUpdatePropertyRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "isPrivate" in value;
    isInstance = isInstance && "categoryId" in value;
    return isInstance;
}
export function UpdatePropertyRequestFromJSON(json) {
    return UpdatePropertyRequestFromJSONTyped(json, false);
}
export function UpdatePropertyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'isPrivate': json['is_private'],
        'categoryId': json['category_id'],
    };
}
export function UpdatePropertyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'is_private': value.isPrivate,
        'category_id': value.categoryId,
    };
}
