// @kinde-oss/kinde-typescript-sdk/dist/models/UpdateCategoryRequest.js
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
import { exists } from '../runtime.js';
/**
 * Check if a given object implements the UpdateCategoryRequest interface.
 */
export function instanceOfUpdateCategoryRequest(value) {
    var isInstance = true;
    return isInstance;
}
export function UpdateCategoryRequestFromJSON(json) {
    return UpdateCategoryRequestFromJSONTyped(json, false);
}
export function UpdateCategoryRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
}
export function UpdateCategoryRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
