// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreatePropertyResponseProperty.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePropertyResponsePropertyToJSON = exports.CreatePropertyResponsePropertyFromJSONTyped = exports.CreatePropertyResponsePropertyFromJSON = exports.instanceOfCreatePropertyResponseProperty = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the CreatePropertyResponseProperty interface.
 */
function instanceOfCreatePropertyResponseProperty(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreatePropertyResponseProperty = instanceOfCreatePropertyResponseProperty;
function CreatePropertyResponsePropertyFromJSON(json) {
    return CreatePropertyResponsePropertyFromJSONTyped(json, false);
}
exports.CreatePropertyResponsePropertyFromJSON = CreatePropertyResponsePropertyFromJSON;
function CreatePropertyResponsePropertyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
    };
}
exports.CreatePropertyResponsePropertyFromJSONTyped = CreatePropertyResponsePropertyFromJSONTyped;
function CreatePropertyResponsePropertyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
    };
}
exports.CreatePropertyResponsePropertyToJSON = CreatePropertyResponsePropertyToJSON;
