// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreatePropertyResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePropertyResponseToJSON = exports.CreatePropertyResponseFromJSONTyped = exports.CreatePropertyResponseFromJSON = exports.instanceOfCreatePropertyResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var CreatePropertyResponseProperty_js_1 = require("./CreatePropertyResponseProperty.js");
/**
 * Check if a given object implements the CreatePropertyResponse interface.
 */
function instanceOfCreatePropertyResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreatePropertyResponse = instanceOfCreatePropertyResponse;
function CreatePropertyResponseFromJSON(json) {
    return CreatePropertyResponseFromJSONTyped(json, false);
}
exports.CreatePropertyResponseFromJSON = CreatePropertyResponseFromJSON;
function CreatePropertyResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'property': !(0, runtime_js_1.exists)(json, 'property') ? undefined : (0, CreatePropertyResponseProperty_js_1.CreatePropertyResponsePropertyFromJSON)(json['property']),
    };
}
exports.CreatePropertyResponseFromJSONTyped = CreatePropertyResponseFromJSONTyped;
function CreatePropertyResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'code': value.code,
        'property': (0, CreatePropertyResponseProperty_js_1.CreatePropertyResponsePropertyToJSON)(value.property),
    };
}
exports.CreatePropertyResponseToJSON = CreatePropertyResponseToJSON;
