// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateCategoryRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCategoryRequestToJSON = exports.CreateCategoryRequestFromJSONTyped = exports.CreateCategoryRequestFromJSON = exports.instanceOfCreateCategoryRequest = exports.CreateCategoryRequestContextEnum = void 0;
/**
 * @export
 */
exports.CreateCategoryRequestContextEnum = {
    Org: 'org',
    Usr: 'usr'
};
/**
 * Check if a given object implements the CreateCategoryRequest interface.
 */
function instanceOfCreateCategoryRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "context" in value;
    return isInstance;
}
exports.instanceOfCreateCategoryRequest = instanceOfCreateCategoryRequest;
function CreateCategoryRequestFromJSON(json) {
    return CreateCategoryRequestFromJSONTyped(json, false);
}
exports.CreateCategoryRequestFromJSON = CreateCategoryRequestFromJSON;
function CreateCategoryRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'context': json['context'],
    };
}
exports.CreateCategoryRequestFromJSONTyped = CreateCategoryRequestFromJSONTyped;
function CreateCategoryRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'context': value.context,
    };
}
exports.CreateCategoryRequestToJSON = CreateCategoryRequestToJSON;
