// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreateCategoryResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCategoryResponseToJSON = exports.CreateCategoryResponseFromJSONTyped = exports.CreateCategoryResponseFromJSON = exports.instanceOfCreateCategoryResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var CreateCategoryResponseCategory_js_1 = require("./CreateCategoryResponseCategory.js");
/**
 * Check if a given object implements the CreateCategoryResponse interface.
 */
function instanceOfCreateCategoryResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfCreateCategoryResponse = instanceOfCreateCategoryResponse;
function CreateCategoryResponseFromJSON(json) {
    return CreateCategoryResponseFromJSONTyped(json, false);
}
exports.CreateCategoryResponseFromJSON = CreateCategoryResponseFromJSON;
function CreateCategoryResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'category': !(0, runtime_js_1.exists)(json, 'category') ? undefined : (0, CreateCategoryResponseCategory_js_1.CreateCategoryResponseCategoryFromJSON)(json['category']),
    };
}
exports.CreateCategoryResponseFromJSONTyped = CreateCategoryResponseFromJSONTyped;
function CreateCategoryResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'message': value.message,
        'code': value.code,
        'category': (0, CreateCategoryResponseCategory_js_1.CreateCategoryResponseCategoryToJSON)(value.category),
    };
}
exports.CreateCategoryResponseToJSON = CreateCategoryResponseToJSON;
