// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/CreatePropertyRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePropertyRequestToJSON = exports.CreatePropertyRequestFromJSONTyped = exports.CreatePropertyRequestFromJSON = exports.instanceOfCreatePropertyRequest = exports.CreatePropertyRequestContextEnum = exports.CreatePropertyRequestTypeEnum = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * @export
 */
exports.CreatePropertyRequestTypeEnum = {
    SingleLineText: 'single_line_text',
    MultiLineText: 'multi_line_text'
};
/**
 * @export
 */
exports.CreatePropertyRequestContextEnum = {
    Org: 'org',
    Usr: 'usr'
};
/**
 * Check if a given object implements the CreatePropertyRequest interface.
 */
function instanceOfCreatePropertyRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "context" in value;
    isInstance = isInstance && "isPrivate" in value;
    isInstance = isInstance && "categoryId" in value;
    return isInstance;
}
exports.instanceOfCreatePropertyRequest = instanceOfCreatePropertyRequest;
function CreatePropertyRequestFromJSON(json) {
    return CreatePropertyRequestFromJSONTyped(json, false);
}
exports.CreatePropertyRequestFromJSON = CreatePropertyRequestFromJSON;
function CreatePropertyRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': json['name'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
        'key': json['key'],
        'type': json['type'],
        'context': json['context'],
        'isPrivate': json['is_private'],
        'categoryId': json['category_id'],
    };
}
exports.CreatePropertyRequestFromJSONTyped = CreatePropertyRequestFromJSONTyped;
function CreatePropertyRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'type': value.type,
        'context': value.context,
        'is_private': value.isPrivate,
        'category_id': value.categoryId,
    };
}
exports.CreatePropertyRequestToJSON = CreatePropertyRequestToJSON;
