// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetCategoriesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetCategoriesResponseToJSON = exports.GetCategoriesResponseFromJSONTyped = exports.GetCategoriesResponseFromJSON = exports.instanceOfGetCategoriesResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var Category_js_1 = require("./Category.js");
/**
 * Check if a given object implements the GetCategoriesResponse interface.
 */
function instanceOfGetCategoriesResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetCategoriesResponse = instanceOfGetCategoriesResponse;
function GetCategoriesResponseFromJSON(json) {
    return GetCategoriesResponseFromJSONTyped(json, false);
}
exports.GetCategoriesResponseFromJSON = GetCategoriesResponseFromJSON;
function GetCategoriesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'categories': !(0, runtime_js_1.exists)(json, 'categories') ? undefined : (json['categories'].map(Category_js_1.CategoryFromJSON)),
        'hasMore': !(0, runtime_js_1.exists)(json, 'has_more') ? undefined : json['has_more'],
    };
}
exports.GetCategoriesResponseFromJSONTyped = GetCategoriesResponseFromJSONTyped;
function GetCategoriesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'categories': value.categories === undefined ? undefined : (value.categories.map(Category_js_1.CategoryToJSON)),
        'has_more': value.hasMore,
    };
}
exports.GetCategoriesResponseToJSON = GetCategoriesResponseToJSON;
