// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetPropertiesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetPropertiesResponseToJSON = exports.GetPropertiesResponseFromJSONTyped = exports.GetPropertiesResponseFromJSON = exports.instanceOfGetPropertiesResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var Property_js_1 = require("./Property.js");
/**
 * Check if a given object implements the GetPropertiesResponse interface.
 */
function instanceOfGetPropertiesResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetPropertiesResponse = instanceOfGetPropertiesResponse;
function GetPropertiesResponseFromJSON(json) {
    return GetPropertiesResponseFromJSONTyped(json, false);
}
exports.GetPropertiesResponseFromJSON = GetPropertiesResponseFromJSON;
function GetPropertiesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'properties': !(0, runtime_js_1.exists)(json, 'properties') ? undefined : (json['properties'].map(Property_js_1.PropertyFromJSON)),
        'hasMore': !(0, runtime_js_1.exists)(json, 'has_more') ? undefined : json['has_more'],
    };
}
exports.GetPropertiesResponseFromJSONTyped = GetPropertiesResponseFromJSONTyped;
function GetPropertiesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'properties': value.properties === undefined ? undefined : (value.properties.map(Property_js_1.PropertyToJSON)),
        'has_more': value.hasMore,
    };
}
exports.GetPropertiesResponseToJSON = GetPropertiesResponseToJSON;
