// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/GetPropertyValuesResponse.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.GetPropertyValuesResponseToJSON = exports.GetPropertyValuesResponseFromJSONTyped = exports.GetPropertyValuesResponseFromJSON = exports.instanceOfGetPropertyValuesResponse = void 0;
var runtime_js_1 = require("../runtime.js");
var PropertyValue_js_1 = require("./PropertyValue.js");
/**
 * Check if a given object implements the GetPropertyValuesResponse interface.
 */
function instanceOfGetPropertyValuesResponse(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfGetPropertyValuesResponse = instanceOfGetPropertyValuesResponse;
function GetPropertyValuesResponseFromJSON(json) {
    return GetPropertyValuesResponseFromJSONTyped(json, false);
}
exports.GetPropertyValuesResponseFromJSON = GetPropertyValuesResponseFromJSON;
function GetPropertyValuesResponseFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'code': !(0, runtime_js_1.exists)(json, 'code') ? undefined : json['code'],
        'message': !(0, runtime_js_1.exists)(json, 'message') ? undefined : json['message'],
        'properties': !(0, runtime_js_1.exists)(json, 'properties') ? undefined : (json['properties'].map(PropertyValue_js_1.PropertyValueFromJSON)),
        'nextToken': !(0, runtime_js_1.exists)(json, 'next_token') ? undefined : json['next_token'],
    };
}
exports.GetPropertyValuesResponseFromJSONTyped = GetPropertyValuesResponseFromJSONTyped;
function GetPropertyValuesResponseToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'code': value.code,
        'message': value.message,
        'properties': value.properties === undefined ? undefined : (value.properties.map(PropertyValue_js_1.PropertyValueToJSON)),
        'next_token': value.nextToken,
    };
}
exports.GetPropertyValuesResponseToJSON = GetPropertyValuesResponseToJSON;
