// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/Property.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyToJSON = exports.PropertyFromJSONTyped = exports.PropertyFromJSON = exports.instanceOfProperty = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the Property interface.
 */
function instanceOfProperty(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfProperty = instanceOfProperty;
function PropertyFromJSON(json) {
    return PropertyFromJSONTyped(json, false);
}
exports.PropertyFromJSON = PropertyFromJSON;
function PropertyFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'isPrivate': !(0, runtime_js_1.exists)(json, 'is_private') ? undefined : json['is_private'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
        'isKindeProperty': !(0, runtime_js_1.exists)(json, 'is_kinde_property') ? undefined : json['is_kinde_property'],
    };
}
exports.PropertyFromJSONTyped = PropertyFromJSONTyped;
function PropertyToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'key': value.key,
        'name': value.name,
        'is_private': value.isPrivate,
        'description': value.description,
        'is_kinde_property': value.isKindeProperty,
    };
}
exports.PropertyToJSON = PropertyToJSON;
