// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/PropertyValue.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PropertyValueToJSON = exports.PropertyValueFromJSONTyped = exports.PropertyValueFromJSON = exports.instanceOfPropertyValue = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the PropertyValue interface.
 */
function instanceOfPropertyValue(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfPropertyValue = instanceOfPropertyValue;
function PropertyValueFromJSON(json) {
    return PropertyValueFromJSONTyped(json, false);
}
exports.PropertyValueFromJSON = PropertyValueFromJSON;
function PropertyValueFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'id': !(0, runtime_js_1.exists)(json, 'id') ? undefined : json['id'],
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_js_1.exists)(json, 'description') ? undefined : json['description'],
        'key': !(0, runtime_js_1.exists)(json, 'key') ? undefined : json['key'],
        'value': !(0, runtime_js_1.exists)(json, 'value') ? undefined : json['value'],
    };
}
exports.PropertyValueFromJSONTyped = PropertyValueFromJSONTyped;
function PropertyValueToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'name': value.name,
        'description': value.description,
        'key': value.key,
        'value': value.value,
    };
}
exports.PropertyValueToJSON = PropertyValueToJSON;
