// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateOrganizationPropertiesRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateOrganizationPropertiesRequestToJSON = exports.UpdateOrganizationPropertiesRequestFromJSONTyped = exports.UpdateOrganizationPropertiesRequestFromJSON = exports.instanceOfUpdateOrganizationPropertiesRequest = void 0;
/**
 * Check if a given object implements the UpdateOrganizationPropertiesRequest interface.
 */
function instanceOfUpdateOrganizationPropertiesRequest(value) {
    var isInstance = true;
    isInstance = isInstance && "properties" in value;
    return isInstance;
}
exports.instanceOfUpdateOrganizationPropertiesRequest = instanceOfUpdateOrganizationPropertiesRequest;
function UpdateOrganizationPropertiesRequestFromJSON(json) {
    return UpdateOrganizationPropertiesRequestFromJSONTyped(json, false);
}
exports.UpdateOrganizationPropertiesRequestFromJSON = UpdateOrganizationPropertiesRequestFromJSON;
function UpdateOrganizationPropertiesRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'properties': json['properties'],
    };
}
exports.UpdateOrganizationPropertiesRequestFromJSONTyped = UpdateOrganizationPropertiesRequestFromJSONTyped;
function UpdateOrganizationPropertiesRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'properties': value.properties,
    };
}
exports.UpdateOrganizationPropertiesRequestToJSON = UpdateOrganizationPropertiesRequestToJSON;
