// @kinde-oss/kinde-typescript-sdk/dist-cjs/models/UpdateCategoryRequest.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Kinde Management API
 * Provides endpoints to manage your Kinde Businesses
 *
 * The version of the OpenAPI document: 1
 * Contact: support@kinde.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateCategoryRequestToJSON = exports.UpdateCategoryRequestFromJSONTyped = exports.UpdateCategoryRequestFromJSON = exports.instanceOfUpdateCategoryRequest = void 0;
var runtime_js_1 = require("../runtime.js");
/**
 * Check if a given object implements the UpdateCategoryRequest interface.
 */
function instanceOfUpdateCategoryRequest(value) {
    var isInstance = true;
    return isInstance;
}
exports.instanceOfUpdateCategoryRequest = instanceOfUpdateCategoryRequest;
function UpdateCategoryRequestFromJSON(json) {
    return UpdateCategoryRequestFromJSONTyped(json, false);
}
exports.UpdateCategoryRequestFromJSON = UpdateCategoryRequestFromJSON;
function UpdateCategoryRequestFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        'name': !(0, runtime_js_1.exists)(json, 'name') ? undefined : json['name'],
    };
}
exports.UpdateCategoryRequestFromJSONTyped = UpdateCategoryRequestFromJSONTyped;
function UpdateCategoryRequestToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
    };
}
exports.UpdateCategoryRequestToJSON = UpdateCategoryRequestToJSON;
